templates/profile/haproxy/frontends.cfg.erb

```diff
@@ -1,16 +1,19 @@
 # Managed by puppet (nebula/profile/haproxy/frontends.cfg.erb)
 
-frontend www-lib-<%= @datacenter -%>-http-front
+<% @frontends.keys.sort.each do |frontend| -%>
+frontend <%= frontend %>-<%= @datacenter -%>-http-front
   bind <%= @floating_ip -%>:80,<%= @networking['ip'] -%>:80
   stats uri /haproxy?stats
-  default_backend www-lib-<%= @datacenter -%>-http-back
+  default_backend <%= frontend %>-<%= @datacenter -%>-http-back
   http-request set-header X-Client-IP %ci
   http-request set-header X-Forwarded-Proto http
 
-frontend www-lib-<%= @datacenter -%>-https-front
+frontend <%= frontend %>-<%= @datacenter -%>-https-front
   bind <%= @floating_ip -%>:443,<%= @networking['ip'] -%>:443 ssl crt /etc/ssl/private/www-lib
```

Comment:
We'll need a separate floating IP for each frontend, and that will have to be configured in keepalived too. We could get around that if we track all the virtual hosts for a particular backend, but I'm not sure I want to do that yet.